azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -1372,7 +1438,7 @@ private void addInitContainers(final ExecutableFlow executableFlow,
             dependency + " in versionSet");
       }
     }
-    if (this.prefetchAllCredentials) {
+    if (this.prefetchAllCredentials && proxyUserList.size() > 0) {
```

Comment:
Why adding the size check? Is it guaranteed that it will always contain at least the submit user?